### PR TITLE
Fix Javadoc URLs for Java 1.8

### PR DESCRIPTION
**Pull request by sanjayl**
Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [x] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [ ] You've updated the readme (if adding/changing middleware)

Thanks!

An issue where the URLs for Javadoc in Java 1.8 has a different format
than prior Java version, namely, parenthesis surrounding argument lists
have been changed to dashes. Added tests as well.

### CODE REVIEW

File: test/clj/cider/nrepl/middleware/info_test.clj [hunk 1]
```diff
@@ -16,6 +16,28 @@
   [x]
   (:resource (info/file-info x)))
 
+(deftest test-javadoc-url
+  (testing "java 1.6"
```
Comment (bbatsov) [inline test/clj/cider/nrepl/middleware/info_test.clj]: We don't support Java 6 anymore (although Clojure itself does), so this test is not necessary. Plus, it's the same behaviour as in 7.

File: src/cider/nrepl/middleware/util/java.clj [hunk 2]
```diff
@@ -73,9 +73,14 @@
        (str/replace "$" ".")
        (str ".html")))
   ([class member argtypes]
-   (str (javadoc-url class) "#" member
-        (when argtypes
-          (str "(" (str/join ",%20" argtypes) ")")))))
+   (let [java-version (read-string util/java-api-version)
```
Comment (Malabarba) [inline src/cider/nrepl/middleware/util/java.clj]: I'd rather use `parse-int` or `Integer.`.

**Comment by sanjayl** [comment]
Made the changes rec'ed above. Added a few more changes to get as close as possible to the new URL spec (although from what I've surmised, we'll never be 100% since the URLs are created via parsing the source, not via reflection). Changed the unit tests a bit.

Apologizing in advance at the ugliness of this code, I wasn't sure how to go about fixing it since the necessary changes were scattered across the code base.

File: src/cider/nrepl/middleware/util/java.clj [hunk 3]
```diff
@@ -73,9 +73,19 @@
        (str/replace "$" ".")
        (str ".html")))
   ([class member argtypes]
-   (str (javadoc-url class) "#" member
-        (when argtypes
-          (str "(" (str/join ",%20" argtypes) ")")))))
+   (let [java-version (Integer/parseInt util/java-api-version)
+         old-format   (< java-version 8)
+         open-paren   (if old-format "(" "-")
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/util/java.clj]: Let's just have two string templates - one for old and one for the new format and insert into them whatever's needed. Or just build the new old or new string without actually introducing those locals. They serve no purpose and their names are confusing for the new format (e.g. close-paren is a "-"). I'd rather have two functions that build the old/new format or something like this.

Comment (Malabarba) [inline src/cider/nrepl/middleware/util/java.clj]: Yes, I think it's better to just have one `if` where each branch is the entire logic for that version.

Comment (bbatsov) [inline src/cider/nrepl/middleware/util/java.clj]: Or one `if` branching out to two helper functions. :-) Anyways, we're on the same page here.

**Comment by sanjayl** [comment]
Yeah, that was pretty dumb of me. Here's the redo

**Comment by bbatsov** [comment]
Perfect! :+1:

**Pull request merged**